### DISCUSSION TITLE
Unable to use awswrangler==2.14.0 in AWS glue python shell within VPC ,private subnet

### DISCUSSION THREAD

**Discussion by sunil-smile** [aws/aws-sdk-pandas#1410] [Q&A] [answered]
Hi Community,

Need your help in resolving the below issue.

I am trying to use awswrangler in my aws glue python shell job and has network connection , which brings the glue job to the private subnet and VPC.

import awswrangler as wr
import boto3

sts = boto3.client('sts')
print(sts.get_caller_identity())

I am just trying to get the AWS account and IAM principal using sts. But it throwing error as **ConnectTimeoutError: Connect timeout on endpoint URL: "https://sts.amazonaws.com/"**

The same glue job without network connections , it works fine.

Kindly assist me , what changes i have to do so then i can use awswrangler within VPC

Regards,
Sunil

**Comment by malachi-constant** [comment]
Does your private subnet have internet egress and/or access to a [private interface endpoint](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_sts_vpce.html) for the AWS STS service?

**Comment by sunil-smile** [comment]
thanks for your response. No it not has internet egress . I think we have private endpoint only to S3. 

Using awsdatawrangler we likes to work in s3 ,athena ,glue

**Comment by sunil-ygroup** [comment, marked as answer]
After setting up the endpoints for Athena , Glue ... I can able to access them using the awswrangler library
